### PR TITLE
Webloc crash

### PR DESCRIPTION
**Pull request by tiennou**
- A straightforward fix for #2176.

- #2020 :

  I'm actually cheating : `-pasteboardByFilteringContents:` doesn't actually do anything since I-cant-even-begin-to-guess-when so I'm just dispatching something that doesn't work to the main thread so it does nothing later™.

  This code was responsible for extracting contents of `.textClipping` files (which ~~are stored in resource forks~~ use dark voodoo magic). I know you could use it to right-arrow into text clippings because I played with that looong ago, but I can't seem to get it to work now. I suspect 64bit ? PPC ? I'm also really not sure what it's expecting to get back anyway... @hmelman does that ring a bell to you ?

- `LSCopyItemInfoForURL` : the commit says it all. I unfortunately got the crashlog only once. I think I got the explanation right though : the autorelease pool would get drained at the end of a scan "cycle", and since I think it does it's think in another thread, you'd get a crash.

### CODE REVIEW

**Comment by pjrobertson** [comment]
@tiennou - indentation... :S
This is the first test my check_indent script has had.

File: Quicksilver/Code-QuickStepCore/QSObject_FileHandling.m [hunk 1]
```diff
@@ -600,8 +600,11 @@ + (QSObject *)fileObjectWithPath:(NSString *)path {
     // initWithArray: deals with file objects that already exist
     QSObject *newObject = [[QSObject alloc] initWithArray:[NSArray arrayWithObject:path]];
     
-	if ([clippingTypes containsObject:[[NSFileManager defaultManager] typeOfFile:path]])
-		[newObject performSelectorOnMainThread:@selector(addContentsOfClipping:) withObject:path waitUntilDone:YES];
+    if ([clippingTypes containsObject:[[NSFileManager defaultManager] typeOfFile:path]]) {
```
Comment (pjrobertson) [inline Quicksilver/Code-QuickStepCore/QSObject_FileHandling.m]: indentation?!

**Comment by tiennou** [comment]
Hope I was quick enough that my failed rebase wasn't visible ;-). Looks like Travis saw me though...
I've made some changes to the check_indent script so it's not as painful. It could be better tough but at least you don't end up doing loads of little commits ;-).

I'll add a check for the CFRelease stuff though.

**Comment by tiennou** [comment]
Yeah, you can also use it locally now, it'll just use `master` as a "default" diff base. It would be nice if it kept the diff header ;-). And then it would be ready to be promoted to commit-hook status ;-).

In case that wasn't clear I added the tab-switching stuff here (Xcode + TextMate).

**Comment by tiennou** [comment]
(Also, w.r.t to your lost comment about thread safety, it's actually `NDResourceFork` that's not threadsafe. I think `pasteboardByFilteringContents:` is actually fine too...

Anyway, right now this code is dead, I've just kept in because 1) I wasn't sure if it was readily fixable, and 2) I'm not even sure what can be done about `NDResourceFork`...

**Comment by hmelman** [comment]
> On Jan 25, 2016, at 8:57 PM, Etienne Samson <notifications@github.com> wrote:
> 
> 	• This code was responsible for extracting contents of .textClipping files (which are stored in resource forks use dark voodoo magic). I know you could use it to right-arrow into text clippings because I played with that looong ago, but I can't seem to get it to work now. I suspect 64bit ? PPC ? I'm also really not sure what it's expecting to get back anyway... @hmelman does that ring a bell to you ?

Not really sure. The Text Manipulation Plugin allows you to right-arrow into a text file and get back a results list with an item for each line. I haven't used it in all long time. Just trying it now, it seems to work right-arrowing into a .txt file.

Howard

**Comment by tiennou** [comment]
Yeah, that's right. I think that resource-forky stuff tries to enable the same behavior with clippings. I might be conflating the two then...

**Comment by pjrobertson** [comment]
@tiennou  - I just cherrypicked your 'smarter script for local deployment' into master.
My first PR since that travis test also just failed because of the same reason :S

Haha!

File: Quicksilver/Code-QuickStepCore/QSObject_Pasteboard.m [hunk 2]
```diff
@@ -80,7 +80,7 @@ - (id)initWithPasteboard:(NSPasteboard *)pasteboard {
 	return [self initWithPasteboard:pasteboard types:nil];
 }
 
-- (void)addContentsOfClipping:(NSString *)path { // Not thread safe?
```
Comment (pjrobertson) [inline Quicksilver/Code-QuickStepCore/QSObject_Pasteboard.m]: why remove the comment if it's true?

**Pull request merged**